### ISSUE TITLE
3.0.0-beta.1 in Addon

### ISSUE THREAD

**Issue by scottkidder** [simplabs/ember-test-selectors#345]
After upgrading an addon to 3.0.0-beta.1 `ember test` breaks with the following:

```
 - stack: TypeError: Cannot read property 'project' of undefined
    at Class._assignOptions (/home/skidder/dev/ember-shared-components/node_modules/ember-test-selectors/index.js:11:18)
    at Class.setupPreprocessorRegistry (/home/skidder/dev/ember-shared-components/node_modules/ember-test-selectors/index.js:31:12)
    at /home/skidder/dev/ember-shared-components/node_modules/ember-cli-preprocess-registry/preprocessors.js:27:13
    at Array.forEach (<anonymous>)
    at setupRegistryForEachAddon (/home/skidder/dev/ember-shared-components/node_modules/ember-cli-preprocess-registry/preprocessors.js:25:10)
    at Object.module.exports.setupRegistry (/home/skidder/dev/ember-shared-components/node_modules/ember-cli-preprocess-registry/preprocessors.js:46:3)
    at Class.init (/home/skidder/dev/ember-shared-components/node_modules/ember-cli/lib/models/addon.js:322:7)
    at Class.superWrapper [as init] (/home/skidder/dev/ember-shared-components/node_modules/core-object/lib/assign-properties.js:34:20)
    at new CoreObject (/home/skidder/dev/ember-shared-components/node_modules/core-object/core-object.js:9:15)
    at new Class (/home/skidder/dev/ember-shared-components/node_modules/core-object/core-object.js:21:5)
```

**Comment by Turbo87** [comment]
ouch, okay, thanks for reporting this. is the addon open-source? do you have any steps to reproduce it?

**Comment by scottkidder** [comment]
Not open source, I can work on a minimal reproduction though.

**Comment by Turbo87** [comment]
that would be great, thanks!

**Comment by scottkidder** [comment]
I've found the root cause to be having `ember-test-selectors` listed as a dependency instead of a devDependency.

**Comment by scottkidder** [comment]
Minimal reproduction here: https://github.com/scottkidder/test-selectors-345

**Comment by scottkidder** [comment]
Hmm, it seems that moving ember-test-selectors to the addon's devDependencies causes the test selectors to be stripped so they cannot be used in a host app.

**Comment by Turbo87** [comment]
@scottkidder are you at EmberConf? would love to talk to you about this use case :)

**Comment by scottkidder** [comment]
@Turbo87 No I wish! Hoping to be able to attend next year.

**Comment by elidupuis** [comment]
FYI, I just tried upgrading to 3.0.0-beta.1 and I'm seeing this same error coming from a couple of my in-repo-addons in the `lib/` directory.

```
Cannot read property 'project' of undefined
```

**Comment by Turbo87** [comment]
I've just spent some time on this, and it seems quite a bit harder to fix than I had hoped :(

more details in https://github.com/ember-cli/ember-cli/pull/7059#issuecomment-480492594

**Comment by Turbo87** [comment]
I've released a fix, or rather workaround, in v3.0.0-beta.2. Please let me know if this solves your issues 🙏

**Comment by elidupuis** [comment]
v3.0.0-beta.2 appears to work in my app where I was seeing the issue before 🚀 

Thanks @Turbo87!
